src/index.js

```diff
@@ -211,46 +28,23 @@ function addMetadata({ types: t }) {
     name: 'addMetadata',
     visitor: {
       Program(babelPath, state) {
-        const GET_TEST_METADATA = 'getTestMetadata';
+        if (!state.opts.enabled) return;
+
         const { filename } = state.file.opts;
         state.opts.shouldLoadFile = shouldLoadFile(filename);
 
-        if (!state.opts.shouldLoadFile) {
-          return;
-        }
-
-        state.opts.transformedModules = [];
-        state.opts.setupCall;
-        state.opts.moduleFunction;
-        state.opts.hooksIdentifier;
-        state.opts.getTestMetadataUID =
-          babelPath.scope.generateUidIdentifier(GET_TEST_METADATA);
+        if (!state.opts.shouldLoadFile) return;
 
-        let importDeclarations = babelPath
-          .get('body')
-          .filter((n) => n.type === 'ImportDeclaration');
-        let emberTestHelpersIndex = importDeclarations.findIndex(
-          (n) => n.get('source').get('value').node === '@ember/test-helpers'
-        );
+        const identifier = t.identifier('getTestMetadata');
+        const importSpecifier = t.importSpecifier(identifier, identifier);
 
-        const getTestMetaDataImportSpecifier = t.importSpecifier(
-          state.opts.getTestMetadataUID,
-          t.identifier(GET_TEST_METADATA)
-        );
-
-        if (emberTestHelpersIndex !== -1) {
-          // Append to existing test-helpers import
-          importDeclarations[emberTestHelpersIndex]
-            .get('body')
-            .container.specifiers.push(getTestMetaDataImportSpecifier);
-        } else {
-          const getTestMetaDataImportDeclaration = t.importDeclaration(
-            [getTestMetaDataImportSpecifier],
+        babelPath.unshiftContainer(
```

Comment:
Do we need to do anything in terms of insuring we're not injecting duplicate imports of `@ember/test-helpers` and/or `getTestMetadata`?

Reply:
we have a test for that case... I'm not 100% sure what the behavior is for duplicate imports, but I assume the second one takes precedence, which is what we want. https://github.com/babel-plugin-ember-test-metadata/babel-plugin-ember-test-metadata/pull/26/files#diff-e6827fdf28f3b2f9dec0318dc552eec4d5db91ed9a7958d5d9048a9d453fe51cR97

Reply:
That might have negative implications for the way modules get resolved in the Ember pipeline. This may be ok for Babel itself, but not so much once all the modules get bundled together and get transpiled into non-ESM module loaders. My point is, the test is not a good indicative of how things get coalesced in an Ember app.

Reply:
sounds good. pushed an update...